client/lib/login/index.js

```diff
@@ -112,3 +113,44 @@ export function getSignupUrl(
 
 	return signupUrl;
 }
+
+export const getLoginLinkPageUrl = ( locale, currentRoute, isGutenboarding ) => {
+	const loginParameters = {
+		locale,
+		twoFactorAuthType: 'link',
+	};
+
+	if ( currentRoute === '/login-in/jetpack' ) {
+		loginParameters.twoFactorAuthType = 'jetpack/link';
+	} else if ( isGutenboarding ) {
+		loginParameters.twoFactorAuthType = 'new/link';
+	}
+
+	return login( loginParameters );
+};
+
+export const canDoMagicLogin = (
+	twoFactorAuthType,
+	oauth2Client,
+	wccomFrom,
+	isJetpackWooCommerceFlow
+) => {
+	if ( ! config.isEnabled( `login/magic-login` ) || twoFactorAuthType ) {
+		return false;
+	}
+
+	// jetpack cloud cannot have users being sent to WordPress.com
+	if ( isJetpackCloudOAuth2Client( oauth2Client ) ) {
+		return false;
+	}
+
+	if ( isWooOAuth2Client( oauth2Client ) && wccomFrom ) {
+		return false;
+	}
+
+	if ( isJetpackWooCommerceFlow ) {
+		return false;
+	}
+
+	return true;
+};
```

Comment:
Same as above https://github.com/Automattic/wp-calypso/pull/53499#discussion_r665457321
Related code that should re-use this logic: https://github.com/Automattic/wp-calypso/blob/update/login-form%2Fwrong-pwd-validation-msg/client/login/wp-login/login-links.jsx#L209-L228

Reply:
Added in 53138a563f81928820b1d81931a4e81e2d65a6f0